From 3e520c9aac3253fe862736f12cdf327d1bdade24 Mon Sep 17 00:00:00 2001
Message-Id: <3e520c9aac3253fe862736f12cdf327d1bdade24.1650636481.git.keller@fs-net.de>
In-Reply-To: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
References: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
From: Zhaoyang Liu <zhaoyang@codeaurora.org>
Date: Tue, 25 Feb 2020 13:33:08 +0800
Subject: [PATCH 28/56] qcacld-2.0: Fix compilation error to support kernel
 version 4.19

In kernel version 4.19 mmc_power_save|restore_host() APIs is gone,
hence add kernel version condition check before using them.

Change-Id: I82970d7ed83ddfd5176f743c74dba48138fe9627
CRs-Fixed: 2628344
---
 CORE/SERVICES/HIF/sdio/linux/native_sdio/src/hif.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/CORE/SERVICES/HIF/sdio/linux/native_sdio/src/hif.c b/CORE/SERVICES/HIF/sdio/linux/native_sdio/src/hif.c
index 4cee00d..bee125a 100644
--- a/CORE/SERVICES/HIF/sdio/linux/native_sdio/src/hif.c
+++ b/CORE/SERVICES/HIF/sdio/linux/native_sdio/src/hif.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013-2019 The Linux Foundation. All rights reserved.
+ * Copyright (c) 2013-2020 The Linux Foundation. All rights reserved.
  *
  * Previously licensed under the ISC license by Qualcomm Atheros, Inc.
  *
@@ -2793,13 +2793,15 @@ static void hif_flush_async_task(HIF_DEVICE *device)
  */
 static int hif_reset_target(HIF_DEVICE *hif_device)
 {
-	int ret;
+	int ret = 0;
 
 	if (!hif_device || !hif_device->func|| !hif_device->func->card) {
 		AR_DEBUG_PRINTF(ATH_DEBUG_ERROR,
 			("AR6000: %s invalid HIF DEVICE \n", __func__));
 		return -ENODEV;
 	}
+
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,19,0))
 	/* Disable sdio func->pull down WLAN_EN-->pull down DAT_2 line */
 	ret = mmc_power_save_host(hif_device->func->card->host);
 	if(ret) {
@@ -2818,6 +2820,7 @@ static int hif_reset_target(HIF_DEVICE *hif_device)
 	}
 
 done:
+#endif
 	return ret;
 }
 
-- 
2.14.5

